### PR TITLE
Update Form Validation for NxCombobox - RSC-1168

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-1168

### CODE REVIEW

File: lib/src/components/NxCombobox/NxCombobox.tsx [hunk 1]
```diff
@@ -51,7 +51,8 @@ function NxComboboxRender<T extends string | number = string>(
       newAttrs = omit(['trimmedValue'], attrs),
 
       isEmpty = !matches.length,
-      isAlert = loading || loadError || isEmpty,
+      showEmptyMessage = isEmpty && value.length,
```
Comment (EmilyBermudez88) [inline lib/src/components/NxCombobox/NxCombobox.tsx:54]: for clarity, why is showEmptyMessage a boolean when empty, and a number when a value is included?

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx [hunk 2]
```diff
@@ -553,17 +613,41 @@ describe('NxCombobox', function() {
...
 
         it('does not set aria-invalid on the combobox', function() {
           expect(quickRender().getByRole('combobox')).not.toHaveAttribute('aria-invalid', 'true');
         });
 
+        describe('when in a form with showValidationErrors', function() {
+          function quickRender(extraProps?: Partial<Props>) {
+            const renderResult = render(
+              <NxForm showValidationErrors onSubmit={() => {}}>
+                <NxCombobox data-testid="combobox" { ...nonValidatableMinimalProps } { ...extraProps } />
```
Comment (EmilyBermudez88) [inline lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx:631]: does the data-testid still need to be here?

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx [hunk 3]
```diff
@@ -593,6 +677,30 @@ describe('NxCombobox', function() {
           expect(multiError.getByRole('combobox')).not.toHaveAttribute('aria-invalid', 'true');
         });
 
+        describe('when in a form with showValidationErrors', function() {
+          function quickRender(extraProps?: Partial<Props>) {
+            const renderResult = render(
+              <NxForm showValidationErrors onSubmit={() => {}}>
+                <NxCombobox data-testid="combobox" { ...nonValidatableMinimalProps } { ...extraProps } />
```
Comment (EmilyBermudez88) [inline lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx:684]: does the data-testid need to be here, same as https://github.com/sonatype/sonatype-react-shared-components/pull/789/files#r976899751?

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx [hunk 4]
```diff
@@ -607,46 +715,105 @@ describe('NxCombobox', function() {
...
 
         it('does not set aria-invalid on the combobox', function() {
           expect(quickRender().getByRole('combobox')).not.toHaveAttribute('aria-invalid', 'true');
         });
 
+        describe('when in a form with showValidationErrors', function() {
+          function quickRender(extraProps?: Partial<Props>) {
+            const renderResult = render(
+              <NxForm showValidationErrors onSubmit={() => {}}>
+                <NxCombobox data-testid="combobox" { ...validatableMinimalProps } { ...extraProps } />
```
Comment (EmilyBermudez88) [inline lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx:733]: same: https://github.com/sonatype/sonatype-react-shared-components/pull/789/files#r976899751

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx [hunk 5]
```diff
@@ -607,46 +715,105 @@ describe('NxCombobox', function() {
...
-          expect(multiError.getByRole('combobox')).toHaveAttribute('aria-invalid', 'true');
+          expect(singleError.getByRole('combobox')).not.toHaveAttribute('aria-invalid', 'true');
+          expect(multiError.getByRole('combobox')).not.toHaveAttribute('aria-invalid', 'true');
         });
 
+        describe('when in a form with showValidationErrors', function() {
+          function quickRender(extraProps?: Partial<Props<string | number>>) {
+            const renderResult = render(
+              <NxForm showValidationErrors onSubmit={() => {}}>
+                <NxCombobox data-testid="combobox" { ...validatableMinimalProps } { ...extraProps } />
```
Comment (EmilyBermudez88) [inline lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx:786]: same as above: https://github.com/sonatype/sonatype-react-shared-components/pull/789/files#r976899751

File: lib/src/components/NxCombobox/NxCombobox.tsx (hunk 1 above)
Comment (rpokorny) [reply lib/src/components/NxCombobox/NxCombobox.tsx:54]: > for clarity, why is showEmptyMessage a boolean when empty, and a number when a value is included?

It's only intended to be used as a boolean-ish value, I just didn't think to `!!` the `value.length`

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx [hunk 6]
```diff
@@ -657,17 +824,41 @@ describe('NxCombobox', function() {
...
 
         it('does not set aria-invalid on the combobox', function() {
           expect(quickRender().getByRole('combobox')).not.toHaveAttribute('aria-invalid', 'true');
         });
 
+        describe('when in a form with showValidationErrors', function() {
+          function quickRender(extraProps?: Partial<Props>) {
+            const renderResult = render(
+              <NxForm showValidationErrors onSubmit={() => {}}>
+                <NxCombobox data-testid="combobox" { ...validatableMinimalProps } { ...extraProps } />
```
Comment (EmilyBermudez88) [inline lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx:842]: same as above: https://github.com/sonatype/sonatype-react-shared-components/pull/789/files#r976899751

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx [hunk 7]
```diff
@@ -697,6 +888,41 @@ describe('NxCombobox', function() {
           expect(multiError.getByRole('combobox')).toHaveAttribute('aria-invalid', 'true');
         });
 
+        describe('when in a form with showValidationErrors', function() {
+          function quickRender(extraProps?: Partial<Props<string | number>>) {
+            const renderResult = render(
+              <NxForm showValidationErrors onSubmit={() => {}}>
+                <NxCombobox data-testid="combobox" { ...validatableMinimalProps } { ...extraProps } />
```
Comment (EmilyBermudez88) [inline lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx:895]: same as above: https://github.com/sonatype/sonatype-react-shared-components/pull/789/files#r976899751

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx (hunk 3 above)
Comment (rpokorny) [reply lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx]: > does the data-testid still need to be here?

No. https://github.com/sonatype/sonatype-react-shared-components/commit/98bd2530253ee6eb992af4a172d2c524ed0e8c85

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx (hunk 4 above)
Comment (rpokorny) [reply lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx]: > does the data-testid still need to be here?

No. https://github.com/sonatype/sonatype-react-shared-components/commit/98bd2530253ee6eb992af4a172d2c524ed0e8c85

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx (hunk 5 above)
Comment (rpokorny) [reply lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx]: > does the data-testid still need to be here?

No. https://github.com/sonatype/sonatype-react-shared-components/commit/98bd2530253ee6eb992af4a172d2c524ed0e8c85

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx (hunk 6 above)
Comment (rpokorny) [reply lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx]: > does the data-testid still need to be here?

No. https://github.com/sonatype/sonatype-react-shared-components/commit/98bd2530253ee6eb992af4a172d2c524ed0e8c85

Generally it's not helpful to comment every instance separately of a similar problem.  A better pattern when you see something like this is to edit your comment on the first instance to indicate that you see a need for the same fix throughout the file.

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx (hunk 7 above)
Comment (rpokorny) [reply lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx]: > does the data-testid still need to be here?

No. https://github.com/sonatype/sonatype-react-shared-components/commit/98bd2530253ee6eb992af4a172d2c524ed0e8c85

File: lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx (hunk 6 above)
Comment (EmilyBermudez88) [reply lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx]: > > does the data-testid still need to be here?
> 
> No. [98bd253](https://github.com/sonatype/sonatype-react-shared-components/commit/98bd2530253ee6eb992af4a172d2c524ed0e8c85)
> 
> Generally it's not helpful to comment every instance separately of a similar problem. A better pattern when you see something like this is to edit your comment on the first instance to indicate that you see a need for the same fix throughout the file.

Sounds good, wasn't sure which was the better approach. I'll edit that next time.

File: lib/src/components/NxCombobox/NxCombobox.tsx (hunk 1 above)
Comment (EmilyBermudez88) [reply lib/src/components/NxCombobox/NxCombobox.tsx:54]: > > for clarity, why is showEmptyMessage a boolean when empty, and a number when a value is included?
> 
> It's only intended to be used as a boolean-ish value, I just didn't think to `!!` the `value.length`

 wanted to confirm if this is good to be left as is, or if it should be changed?

Comment (rpokorny) [reply lib/src/components/NxCombobox/NxCombobox.tsx:54]: > wanted to confirm if this is good to be left as is, or if it should be changed?

I'm ok with it, but can change it if you or someone else wants it changed.

Comment (EmilyBermudez88) [reply lib/src/components/NxCombobox/NxCombobox.tsx:54]: > > wanted to confirm if this is good to be left as is, or if it should be changed?
> 
> I'm ok with it, but can change it if you or someone else wants it changed.

I'm ok with it, just wanted to confirm.

**Review by EmilyBermudez88 — APPROVED** [review]

**Pull request merged**